pkg/apis/trust/v1alpha1/types_bundle.go

```diff
@@ -127,6 +138,12 @@ type BundleStatus struct {
 	// Known condition types are `Bundle`.
 	// +optional
 	Conditions []BundleCondition `json:"conditions,omitempty"`
+
+	// DefaultCAPackageVersion, if set and non-empty, indicates the version information
+	// which was retrieved when the set of default CAs was requested in the bundle
+	// source. This should only be set if useDefaultCAs was set to "true" on a source,
+	// and will be the same for the same version of a bundle with identical certificates.
+	DefaultCAPackageVersion *string `json:"defaultCAVersion,omitempty"`
```

Comment:
I'm not sure about reporting this version here, we also don't do this for secret/ configmap sources; would it make sense to just remove this status field?

Reply:
No, this is absolutely required IMO and was added in response to @JoshVanL 's previous review [here](https://github.com/cert-manager/trust-manager/pull/46#discussion_r1049753380).

The question this answers is "what version of the source was used when creating the target?"

For a ConfigMap / Secret, you can always look at the configmap or secret to find out the contents. In fact, trust-manager can't help with that - it doesn't know what the contents are until it reads for itself and doesn't store the contents once the target is written.

For InLine you look at the Bundle resource `spec` directly.

For DefaultCAs there's no easy way to answer that question without this status field. The other source would either be trust-manager logs or by inspecting the way trust-manager was set up (e.g. by looking at values which were set when the helm chart was installed).

We need a single place where someone with access to read a Bundle resource can look to determine whether that Bundle's DefaultCAs are out of date because it's critical that when the defaultCAs are out of date, they can be updated quickly. I think this status field is the best place for that.

Reply:
I mean we could include the generation of the secret configmap and the resource itself in the status.

Reply:
Also, out of curiosity: when would there be a situation in which the secret defaultCAs are not up-to-date? Why would the defaultCAs be out-of-date and the secret/ configmap not?

Reply:
> I mean we could include the Kubernetes Resource Generation of the secret configmap and the resource itself in the status.

We could, but it's significantly less valuable than adding the defaultCAs version here. I can see how it'd help diagnose if a Secret/CM was updated but the Bundle which references that Secret/CM didn't pick up the changed source. But that would be a feature for a different PR in any case.

> Also, out of curiosity: when would there be a situation in which the secret defaultCAs are not up-to-date? Why would the defaultCAs be out-of-date and the secret/ configmap not?

Secrets/CMs/InLine are all user-controller and user-specified. They're out of date if the user says they're out of date. We can't make that call - a Bundle uses whatever it's pointed at. If the user updates a Secret/CM/InLine, then the Bundle controller should update ASAP with the latest available update.

DefaultCAs are different - they're configured at the time when trust-manager is installed (likely via helm's `defaultPackageImage.tag`). They're out of date if there's a new version of the default package available at https://quay.io/repository/jetstack/cert-manager-package-debian and there are only new versions published there if Debian publish new versions of their `ca-certificates` package.

Put another way: The update cycle is different. For Secret/CM/InLine, users update some Secret/CM/InLine in their cluster. For defaultCAs, users update trust-manager to point at a new default CA package.

Reply:
@SgtCoDFish can you consider this alternative approach and let me know what you think:
https://github.com/SgtCoDFish/cert-manager-trust-manager/pull/1
(I would prefer not using a new struct field specifically for reporting this one version)